### PR TITLE
kloak: init at v0.2

### PR DESCRIPTION
**Pull request by ryneeverett**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kloak/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchgit }:
+
+stdenv.mkDerivation {
+  pname = "kloak";
+  version = "v0.2";
+
+  src = fetchgit {
```
Comment (mweinelt) [inline pkgs/os-specific/linux/kloak/default.nix:7]: Use fetchFromGitHub instead.

https://nixos.org/nixpkgs/manual/#chap-pkgs-fetchers

File: pkgs/os-specific/linux/kloak/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    url = "https://github.com/vmonaco/kloak.git";
+    rev = "49ca7208f15b976b51df23666667f14c63a631d7";
```
Comment (mweinelt) [inline pkgs/os-specific/linux/kloak/default.nix:9]:
```suggestion
    rev = "${version}";
```

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by ryneeverett** [comment]
Thanks for the review @mweinelt!

I wasn't sure if the website-specific fetchers were preferred but I'm happy to defer to your judgement.

File: pkgs/os-specific/linux/kloak/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "kloak";
+  version = "v0.2";
```
Comment (mweinelt) [inline pkgs/os-specific/linux/kloak/default.nix:5]: Please move the `v` prefix into `src.rev` instead.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/kloak/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./kloak ./eventcap $out/bin/
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Keystroke-level online anonymization kernel: obfuscates typing behavior at the device level";
+    longDescription = "Note that this requires loading the 'uinput' kernel module.";
```
Comment (mweinelt) [inline pkgs/os-specific/linux/kloak/default.nix:21]: I'm not sure that is an appropriate use of the longDescription field.

Comment (ryneeverett) [reply pkgs/os-specific/linux/kloak/default.nix:21]: Perhaps not though I've seen basic usage notes in this field for other packages. Would a preferred alternative be a wrapper script to echo a warning if this kernel module is not loaded?

Comment (mweinelt) [reply pkgs/os-specific/linux/kloak/default.nix:21]: I neither have a strong opinion nor much experience and would like to defer to someone else.

**Comment by AndersonTorres** [comment]
Closing as dead.

**Pull request closed without merging**